packages/website/docs/docs/formatting-toolbar.md

```diff
@@ -18,9 +18,180 @@ The Formatting Toolbar appears whenever you highlight text in the editor, and is
+        onClick={() => {
+          props.editor.toggleStyles({
+            textColor: "blue",
+            backgroundColor: "blue",
+          });
+        }}
+        isSelected={
+          props.editor.getActiveStyles().textColor === "blue" &&
+          props.editor.getActiveStyles().backgroundColor === "blue"
+        }
+      >
+        Blue
+      </ToolbarButton>
+    </Toolbar>
+  );
+};
+```
+
+The components used to build this custom Formatting Toolbar are pre-made and come with BlockNote. These can be used if you need to change the functionality or layout, but want to keep the default styling. You can find all the pre-made components BlockNote offers in the [Default Items](/docs/formatting-toolbar#default-items) and [Custom Items](/docs/formatting-toolbar#custom-items) subsections.
+
+However, you could render whatever you like in `CustomFormattingToolbar`, allowing for fully custom designs.
+
+Also, if you're unsure about what's happening inside `ToolbarButton`'s `onClick` and `isSelected` props, head to [Introduction to Blocks](/docs/blocks), which will guide you through reading & manipulating blocks in the editor using code.
+
+After creating a custom toolbar and editor instance, we can add it to the editor by rendering it inside `BlockNoteView`:
+
+```jsx
+export default function App() {
+  // Creates a new editor instance.
+  const editor: BlockNoteEditor = useBlockNote({});
+  
+  // Renders the editor instance.
+  return (
+    <BlockNoteView editor={editor}>
+      <FormattingToolbarPositioner
+        editor={editor}
+        formattingToolbar={CustomFormattingToolbar}
+      />
+      <HyperlinkToolbarPositioner editor={editor} />
+      <SlashMenuPositioner editor={editor} />
+      <SideMenuPositioner editor={editor} />
+    </BlockNoteView>
+  );
+}
+```
+
+There's a bit more we have to explain here though. First of all, let's talk about the `FormattingToolbarPositioner` component. Not only can you change what's inside the Formatting Toolbar and how it looks, but also when it's shown/hidden, as well as where it's positioned.
+
+If you're looking to customize this to create, for example, a static Formatting Toolbar, head to [Custom Formatting Toolbar Positioner](/docs/formatting-toolbar#custom-formatting-toolbar-positioner). If you only want to change the toolbar element itself, you can use the `FormattingToolbarPositioner` to apply the default behaviour.
+
+TODO: Text below is not final. We should change how we remove & customize the menus & toolbars. Take the scenarios:
+- Customizing 1 UI element
+- Customizing all UI elements
+- Removing 1 UI element
+- Removing all UI elements
+
+By default, if you don't provide any children the `BlockNoteView`, BlockNote uses the default UI. However, if you want to provide a custom Formatting Toolbar, you should also add the default positioners for them (default components are used if custom ones aren't provided), as BlockNote won't render them otherwise.
```

Comment:
I think we should create a separate page about BlockNoteView, that explains the defaults